src/storage/store/list_chunk_data.cpp

```diff
@@ -157,6 +157,25 @@ void ListChunkData::appendNullList() {
     nullData->setNull(appendPosition, true);
 }
 
+void ListChunkData::scan(ValueVector& output, offset_t offset, length_t length) const {
+    KU_ASSERT(offset + length <= numValues);
+    if (nullData) {
+        output.setNullFromBits(nullData->getNullData()->getNullMask().getData(), offset,
+            0 /*dstOffset*/, length);
+    }
+    auto dataOffset = getListStartOffset(offset);
+    auto dataSize = 0ul;
```

Comment:
I think there is one missing part: list start offsets can be un-sorted. this is due to updating a list in the middle of the chunk (say the chunk has 100 values, dataColumnChunk has 1000 values. we update a list value at pos 10, we append new list data to the end of the dataColumnChunk, and update the offset at pos 10 pointing to new pos at dataColumnChunk, which would be 1000). If unsorted, we should fall back to read the list value one at a time. The logic should roughly be same as here  https://github.com/kuzudb/kuzu/blob/96e9f94b8a8fc398974084b2e91afdd0fca3ff27/src/storage/store/list_column.cpp#L238